tests/rule_based_profiler/test_rule_based_profiler.py

```diff
@@ -1132,6 +1163,36 @@ def test_get_profiler(
     assert isinstance(profiler, RuleBasedProfiler)
 
 
+@mock.patch("great_expectations.data_context.data_context.DataContext")
+def test_save_profiler(
+    mock_data_context: mock.MagicMock,
+    populated_profiler_store: ProfilerStore,
+    profiler_config_with_placeholder_args: RuleBasedProfilerConfig,
+):
+    with mock.patch(
+        "great_expectations.data_context.store.profiler_store.ProfilerStore.set",
+        return_value=profiler_config_with_placeholder_args,
+    ):
+        mock_data_context.save_profiler(
+            profiler=profiler_config_with_placeholder_args,
+            profiler_store=populated_profiler_store,
+            name="my_profiler",
+            ge_cloud_id=None,
+        )
+
+    with mock.patch(
+        "great_expectations.data_context.store.profiler_store.ProfilerStore.get",
+        return_value=profiler_config_with_placeholder_args,
+    ):
+        profiler = RuleBasedProfiler.get_profiler(
+            data_context=mock_data_context,
+            profiler_store=populated_profiler_store,
+            name="my_profiler",
+            ge_cloud_id=None,
+        )
+    assert isinstance(profiler, RuleBasedProfiler)
+
+
```

Comment:
If this is a context level method, I think we should move it to the context tests right?

What is it that we're aiming to test here? Given a profiler and name/ge_cloud_id, we can determine a key and use it to store the config on that profiler object.

As such, I think we might want to move away from the mocks we have here.

Reply:
since the store deals with filesystem interaction, I think it may make sense to actually mock that object? we can then assert that `set()` was invoked from the store with the appropriate config arg? I think there are a few ways to go about getting coverage.

As this is designed to be a unit test, I'd like us to isolate just the `save` method; could we remove the call to `get_profiler`?